engine/src/main/java/org/terasology/telemetry/TelemetryScreen.java

```diff
@@ -0,0 +1,144 @@
+import org.terasology.module.DependencyResolver;
+import org.terasology.module.Module;
+import org.terasology.module.ModuleEnvironment;
+import org.terasology.module.ResolutionResult;
+import org.terasology.module.predicates.FromModule;
+import org.terasology.naming.Name;
+import org.terasology.registry.In;
+import org.terasology.rendering.nui.CoreScreenLayer;
+import org.terasology.rendering.nui.WidgetUtil;
+import org.terasology.rendering.nui.animation.MenuAnimationSystems;
+import org.terasology.rendering.nui.databinding.BindHelper;
+import org.terasology.rendering.nui.layouts.ColumnLayout;
+import org.terasology.rendering.nui.layouts.RowLayout;
+import org.terasology.rendering.nui.layouts.ScrollableArea;
+import org.terasology.rendering.nui.widgets.UILabel;
+import org.terasology.telemetry.metrics.Metric;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The metric menu.
+ */
+public class TelemetryScreen extends CoreScreenLayer {
+
+    public static final ResourceUrn ASSET_URI = new ResourceUrn("engine:telemetryScreen");
+
+    @In
+    private Config config;
+
+    @In
+    private ModuleManager moduleManager;
+
+    @In
+    private TranslationSystem translationSystem;
+
+    @In
+    private Metrics metric;
+
+    private static final Logger logger = LoggerFactory.getLogger(TelemetryScreen.class);
+
+
+    private final int horizontalSpacing = 12;
+
+    @Override
+    public void initialise() {
+        setAnimationSystem(MenuAnimationSystems.createDefaultSwipeAnimation());
+        ColumnLayout mainLayout = new ColumnLayout();
+        mainLayout.setHorizontalSpacing(8);
+        mainLayout.setVerticalSpacing(8);
+
+        Map<TelemetryCategory, Class> telemetryCategories = Maps.newHashMap();
+
+        DependencyResolver resolver = new DependencyResolver(moduleManager.getRegistry());
+        for (Name moduleId : moduleManager.getRegistry().getModuleIds()) {
+            Module module = moduleManager.getRegistry().getLatestModuleVersion(moduleId);
+            if (module.isCodeModule()) {
+                ResolutionResult result = resolver.resolve(moduleId);
+                if (result.isSuccess()) {
+                    try (ModuleEnvironment environment = moduleManager.loadEnvironment(result.getModules(), false)) {
+                        for (Class<?> holdingType : environment.getTypesAnnotatedWith(TelemetryCategory.class, new FromModule(environment, moduleId))) {
+                            TelemetryCategory telemetryCategory = holdingType.getAnnotation(TelemetryCategory.class);
+                            telemetryCategories.put(telemetryCategory, holdingType);
+                        }
+                    }
+                }
+            }
+        }
+
+        for (TelemetryCategory telemetryCategory: telemetryCategories.keySet()) {
+            Class metricClass = telemetryCategories.get(telemetryCategory);
```

Comment:
if you iterate a keySet and directly fetch the value for a key, you could replace this with an iteration over the entrySet instead.